deploy/cloudflare.md

```diff
@@ -11,29 +11,38 @@ outline: deep
 
 Uploading via Web UI is limited to 300 MB.
 
-Use [rclone](https://rclone.org/downloads/) to upload larger PMTiles archives to R2.
+Use [rclone](https://rclone.org/downloads/) to upload larger PMTiles archives to R2 (you can use the `rclone/rclone` docker image in order to avoid installation, the config is at `/etc/rclone` for mounting). 
+In order to use rclone you'll need to create an API key. Copy the "Access Key ID", the "Secret Access Key" and the "Endpoints for S3 clients" from the API key creation screen.
+Run the following commands:
+1. `rclone config` and follow the on screen questions, note that the endpoint should be `https://<ACCOUNT_ID>.r2.cloudflarestorage.com/<BUCKET_NAME>`
```

Comment:
The endpoint should not have the `/<BUCKET_NAME` at the end.

Reply:
It didn't work for me if I didn't add it.

Reply:
you need to treat the bucket name as the folder name in `rclone copyto ENDPOINTNAME:BUCKETNAME/foo.pmtiles`

Reply:
I tried that too, got 403... :-/

Reply:
This is what I'm getting:
```
docker run -i -v ~/workspace/:/config/rclone/ rclone/rclone copy -v /config/rclone/yarn.lock Cloudflare:my-bucket/
2023/12/26 15:45:52 ERROR : yarn.lock: Failed to copy: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id: 
2023/12/26 15:45:52 ERROR : Attempt 1/3 failed with 1 errors and: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id: 
2023/12/26 15:45:52 ERROR : yarn.lock: Failed to copy: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id: 
2023/12/26 15:45:52 ERROR : Attempt 2/3 failed with 1 errors and: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id: 
2023/12/26 15:45:52 ERROR : yarn.lock: Failed to copy: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id: 
2023/12/26 15:45:52 ERROR : Attempt 3/3 failed with 1 errors and: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id: 
2023/12/26 15:45:52 INFO  : 
Transferred:              0 B / 0 B, -, 0 B/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         1.6s

2023/12/26 15:45:52 Failed to copy: failed to prepare upload: AccessDenied: Access Denied
        status code: 403, request id: , host id:
```

Reply:
What about

`rclone/rclone copy -v /config/rclone/yarn.lock Cloudflare:my-bucket/yarn.lock`

Reply:
I'll try, but when I did that with the bucket in the endpoint it created a folder with yarn lock name and placed the file inside it...

Reply:
Same error, it didn't create a folder but didn't upload the file as well...

Reply:
To confirm, when you look at your configured rclone endpoint in `~/.config/rclone/rclone.conf` it should have an entry like 

```
[Cloudflare]
...
endpoint = https://ACCOUNT.r2.cloudflarestorage.com
```

Reply:
Yup, if I have this configured like that and without the bucket name in the endpoint I get a 403 error.

Reply:
I also tried with version 1.65 of the docker container, and also downloaded and install version 1.65 locally.
I get the same error message.
The token/keys that I have provided to the rclone has the permissions of manipulating files but not the permissions of creating buckets etc...
In any case, I'll be moving on, feel free to merge this and update the relevant parts, this is my experience. Yours might be different, and I might have misconfigured some things, but I guess I won't be the only one encountering those issues, so it's worth documenting some of it I guess...

Reply:
I haven't received any other reports of this; you are reporting that with that existing credentials configuration you can upload to a folder, but not to the root? otherwise I would suspect there is some misconfiguration in the credentials.

In any case the choice of `rclone` here is just because that is one popular tool, but you can use any of dozens of libraries/CLIs to upload a file to the root directory of the bucket. Is there a different tool you prefer?

Reply:
Reading the docs of cloud flare, they suggest wrangler for smaller files (which I guess the web UI is more intuitive and easier) and rclone for larger files.
I haven't tried any other tool and I'm not very familiar with these kind of things and this is my first time trying to upload a large file to R2 in cloud flare.

It's not super important for me to add the file to the root as it does work inside a folder using all the other steps, only the address has the folder and is not using the root, which I personally don't mind, it's just a URL...

Since I was not able to upload to the root I don't know how to change the docs to reflect that, besides maybe adding a disclaimer that uploading to the root may fail, and in that case upload to a folder...

Reply:
This documentation is about serving pmtiles through cloudflare workers, and having the files in the root of the bucket is the simplest example. We can't change the documentation to suggest a more complex configuration that implies the open source code in this project only works with folders. 

I think there is potentially some usability improvements to `rclone` to make the error messages more helpful regarding configuration, but those need to be filed to the RClone project and not this one.

Reply:
I agree, but the bottom line, is that I can't follow the current instructions (before this PR).
I would love to place my pmtiles at the root, but I was unable to do it no matter what I tried.
Yes, I might be the only one, but I doubt it.
I'm not saying rclone is not buggy, it probably is, and I might go and open an issue there, but until this is solved I believe a disclaimer is in place.
I'll update the docs and you decide if it's good enough or not.

Reply:
I've updated the docs to something that never worked for me. I hope it will work for others. I also added a disclaimer/potential workaround.

Reply:
Can you try the command again with the flag ` --s3-no-check-bucket`?

Reply:
Sure, give me a sec and I'll try.

Reply:
WOW! This worked! Thanks! I'll update the docs.

Reply:
I have updated the docs

Reply:
I had the environment variable `export RCLONE_S3_NO_CHECK_BUCKET=true` so that was what made it work on my end. Thanks for debugging, merged.